tests/conftest.py

```diff
@@ -36,3 +37,22 @@ def api_manager() -> ApiManager:
     api_manager_.reset()
     yield api_manager_
     api_manager_.__dict__.update(old_attrs)
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--use-api-key",
+        action="store_true",
+        default=False,
+        help="Use api key for the tests that need an api key AND whose behavior changes with your code. This will save Open AI answers for future use.",
+    )
+
+
+def pytest_configure(config):
+    pytest.use_api_key = config.getoption("--use-api-key")
+
+
+def pytest_collection_modifyitems(config, items):
+    for item in items:
+        if getattr(item.obj, "requires_api_key", False):
+            item.add_marker(pytest.mark.xfail(run=False))
```

Comment:
Why are we marking `xfail` instead of `skip` with a reason?.  My understanding of how this decorator works is that it will take any test marked `requires_api_key`, mark it xfail, and never run it under any circumstances.

Reply:
It looks like we are marking them xfail both here and in the `requires_api_key` decorator, but with different paramterization, which is extremely confusing and hard to predict.